# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/printing/nsPrintJob.cpp
# Commit: cc4de7b25404
# Full Hash: cc4de7b25404f927d2219f81000d0e0f5c9f7cfe
# Author: Jonathan Watt <jwatt@jwatt.org>
# Date: 2022-05-23 09:52:44
# Regressor Bug: 1770539
# File Overlap Count: 1
# Description:
#   Bug 1770539 p4 - Move nsPrintData::mNumPrintablePages to nsPrintJob. r=emilio
#   
#   Depends on D146975
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D146976
# ==============================================================================

diff -r f946921b9173 -r cc4de7b25404 layout/printing/nsPrintJob.cpp
--- a/layout/printing/nsPrintJob.cpp	Sun May 22 09:37:56 2022 +0000
+++ b/layout/printing/nsPrintJob.cpp	Sun May 22 09:37:56 2022 +0000
@@ -876,9 +876,9 @@
   PR_PL(("-------------------------------------------------------\n"));
   PR_PL(("\n"));
 
-  CalcNumPrintablePages(printData->mNumPrintablePages);
+  CalcNumPrintablePages(mNumPrintablePages);
 
-  PR_PL(("--- Printing %d pages\n", printData->mNumPrintablePages));
+  PR_PL(("--- Printing %d pages\n", mNumPrintablePages));
   DUMP_DOC_TREELAYOUT;
 
   // Print listener setup...
@@ -905,14 +905,13 @@
                         docURLStr);
 
   int32_t startPage = 1;
-  int32_t endPage = printData->mNumPrintablePages;
+  int32_t endPage = mNumPrintablePages;
 
   nsTArray<int32_t> ranges;
   mPrintSettings->GetPageRanges(ranges);
   for (size_t i = 0; i < ranges.Length(); i += 2) {
     startPage = std::max(1, std::min(startPage, ranges[i]));
-    endPage = std::min(printData->mNumPrintablePages,
-                       std::max(endPage, ranges[i + 1]));
+    endPage = std::min(mNumPrintablePages, std::max(endPage, ranges[i + 1]));
   }
 
   nsresult rv = NS_OK;